lib/ui/painting.dart

```diff
@@ -42,6 +42,9 @@ bool _offsetIsValid(Offset offset) {
 bool _matrix4IsValid(Float64List matrix4) {
   assert(matrix4 != null, 'Matrix4 argument was null.');
   assert(matrix4.length == 16, 'Matrix4 must have 16 entries.');
+  assert(() {
+    return matrix4.every((double value) => value.isFinite);
+  }(), 'Matrix4 entries must be finite.');
   return true;
```

Comment:
You want to perform validation in profile and release modes too. How is passing non-finite matrices to the engine in release mode OK?

Reply:
I'm closing this in favor of @liyuqian's, but we're already doing this all over the place.  Everywhere we assert that a offset, rect, or matrix is valid it's only in debug mode.